proposals/NNNN-pack-destructuring-and-pack-splitting.md

```diff
@@ -0,0 +1,183 @@
+}
+```
+
+## Technical Design
+
+### Grammar Extension  
+```ebnf
+tuple-pattern-element → pattern
+                      | 'repeat' 'each' identifier
+tuple-expression-element → expression
+                         | 'repeat' 'each' pack-expression
+```
+
+### AST & Type System  
+- Introduce `PackExpansionPattern` and `PackExpansionExpr` nodes.  
+- Enforce at most one `repeat each` at each tuple nesting level.  
+- Match arity at compile time; no runtime checks needed.
+
+### Source Compatibility  
+This feature is purely additive. Existing code—packs, expansions, and overload resolution—continues to compile without change.
+
+### ABI Stability  
+No new runtime structures or calling conventions are introduced. All destructuring is compiled to existing tuple projection instructions.
+
+### Performance  
+Compile‑time type checking adds negligible cost. Generated code for tuple access and calls is identical to hand‑written projections.
+
+## Argument Binding Rules  
+1. Bind all non‑pack parameters in declaration order.  
+2. The single `repeat each` parameter then consumes the remaining arguments as one pack.  
+3. Extra or missing arguments produce diagnostics as usual.
+
+```swift
+func accept(a: Int, b: String, c: repeat each C) { … }
+
+// Valid:
+accept(repeat each (1, "a", true, false))
+// a = 1, b = "a", c = (true, false)
+
+// Error: too few args
+accept(repeat each (1, "a"))
+```
+
+## Impact on Standard Library & Real‑World Examples
+
+### Combine.zip Before  
+```swift
+// 10+ overloads:
+func zip<A,B>(_ a: A, _ b: B) -> Zip2<A,B>
+func zip<A,B,C>(_ a: A, _ b: B, _ c: C) -> Zip3<A,B,C>
+// … up to Zip10
+```
+
+### Combine.zip After  
+```swift
+struct Zip<repeat each S>: Publisher {
+  typealias Output = (repeat (each S).Output)
+
+  let publishers: (repeat each S)
+
+  func receive<Sub: Subscriber>(subscriber: Sub) where Sub.Input == Output {
+    let (first, repeat each rest) = publishers
+    first.receive(ZipHelper(subscriber, rest))
+  }
+}
+```
+
+**Result**: ~250 LOC → ~20 LOC, no fixed arity limit, compile‑time safety preserved.
+
+## Future Directions  
+- **Multi‑Pack Operations**  
+  ```swift
+  func zip(_ s1: repeat each S1, _ s2: repeat each S2) -> (repeat each (S1, S2))
+  ```
+- **Named Pack Elements**  
+  ```swift
+  let (header: repeat each headers, body: repeat each bodies) = httpResponse
+  ```
+- **Variadic async/await**  
+  ```swift
+  async let (firstResult, repeat each partials) = fetchAll(repeat each requests)
+  ```
+
+## Alternatives Considered  
+
+### 1. Fixed‑Arity Overloads  
+- Pros: trivial, no language changes  
+- Cons: boilerplate, arbitrary limits, code bloat  
+
+
+## References  
+1. [SE‑0393: Parameter Packs](https://github.com/apple/swift-evolution/blob/main/proposals/0393-parameter-packs.md) – Introduces variadic type‐parameter packs.  
+2. [SE‑0404: Pack Iteration](https://github.com/apple/swift-evolution/blob/main/proposals/0404-pack-iteration.md) – Enables iteration over packs.  
```

Comment:
Pack Iteration is proposal 408 not 404. You do get a 404 when you click that link though.
```suggestion
2. [SE‑0404: Pack Iteration](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0408-pack-iteration.md) – Enables iteration over packs.  
```